articles/azure-functions/functions-dotnet-class-library.md

```diff
@@ -295,29 +295,53 @@ You can't use `out` parameters in async functions. For output bindings, use the
 
 ```csharp
-public static class CancellationTokenExample
+using Microsoft.Azure.ServiceBus;
+using System.Threading;
+
+namespace ServiceBusCancellationToken
 {
-    public static void Run(
-        [QueueTrigger("inputqueue")] string inputText,
-        TextWriter logger,
-        CancellationToken token)
+    public static class servicebus
     {
-        for (int i = 0; i < 100; i++)
+        [FunctionName("servicebus")]
+        public static void Run([ServiceBusTrigger("csharpguitar", Connection = "SB_CONN")]
+               Message[] messages, CancellationToken cancellationToken, ILogger log)
         {
-            if (token.IsCancellationRequested)
+            try
+            { 
+                foreach (var message in messages)
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        log.LogInformation("A cancellation token was received. Taking precautionary actions.");
+                        //Take precautions like noting how far along you are with processing the batch
+                        log.LogInformation("Precautionary activities --complete--.");
+                        break;
+                    }
+                    else
+                    {
+                        //business logic as usual
+                        log.LogInformation($"Message: {message} was processed.");
+                    }
+                }
+            }
+            catch (Exception ex)
             {
-                logger.WriteLine("Function was cancelled at iteration {0}", i);
-                break;
+                log.LogInformation($"Something unexpected happened: {ex.Message}");
             }
-            Thread.Sleep(5000);
-            logger.WriteLine("Normal processing for queue message={0}", inputText);
         }
     }
 }
 ```
+A common pattern for iterating through an array uses the foreach statement.   Within the foreach statement, prior to processing the message, check if a cancellation token has been generated.  The most common scenario will be that the IsCancellationRequested property is false and, in that case, your coded business logic will execute.  In a seldom, but realistic scenario that IsCancellationRequested is true, you will need to take some precautionary actions.  For example, writing a log which stores the fact that this happened and perhaps store the portion of the message batch which has not yet been processed.  If you do the latter, then your startup code needs to check if there are any message batches which experienced this shutdown procedure.  The actual requirement is based on your specific scenario.
```

Comment:
Edits and formatting suggestions...note that the issue of persistence in Functions is a bit tough for stateless functions. Should we recommend that they write state info to their Storage account?
```suggestion
As in the previous example, you commonly iterate through an array using a `foreach` loop.  Within this loop and before processing the message, you should check the value of `cancellationToken.IsCancellationRequested` to see if cancellation is pending. In the case where `IsCancellationRequested` is `true`, you may need to take some actions to prepare for a graceful shutdown . For example, you might want to log the status of your code before the shutdown or perhaps write to a persisted store the portion of the message batch which hasn't yet been processed.  If you write this kind of information to a persisted store, then your startup code needs to check the store for any unprocessed message batches written during shutdown. What your code needs to do during graceful shutdown depends on your specific scenario.
```